src/Microsoft.OData.Edm/Validation/EdmError.cs

```diff
@@ -25,12 +25,31 @@ public EdmError(EdmLocation errorLocation, EdmErrorCode errorCode, string errorM
             this.ErrorLocation = errorLocation;
             this.ErrorCode = errorCode;
             this.ErrorMessage = errorMessage;
+            this.ErrorLevel = "error";
         }
 
+        /// <summary>
+        /// Initializes a new instance of the EdmError class.
+        /// </summary>
+        /// <param name="errorLocation">The location where the error occurred.</param>
+        /// <param name="errorCode">An integer code representing the error.</param>
+        /// <param name="errorMessage">A human readable message describing the error.</param>
+        /// <param name="errorLevel">A string describing the error severity.</param>
+        public EdmError(EdmLocation errorLocation, EdmErrorCode errorCode, string errorMessage, string errorLevel)

```

Comment:
@wachugamaina Why not have this constructor call the existing one that has less parameters?
```
this(errorLocation, errorCode, errorMessage);
this.ErrorLevel = errorLevel;
```

Reply:
Because the constructor with the fewer parameters sets the ErrorLevel to the default error.

Reply:
@wachugamaina The proposal from @gathogojr is a better design. You do not need to set default error level to `Error `as well. The constructor is already being used as is no need to interfere with it.

Reply:
@wachugamaina I'd expect error and error level to represent different things. Plus, calling the other constructor doesn't inhibit you from setting the `ErrorLevel` to the value passed into the constructor. The first line calls the other constructor then second sets the `ErrorLevel`

Reply:
Noted @gathogojr